next/utils/newBooksService.ts

```diff
@@ -0,0 +1,164 @@
+
+interface OpacBook {
+  title: { _text: string }
+  author: { _text: string }
+  coverURL: { _text: string }
+  recURL: { _text: string }
+}
+
+interface OpacResponse {
+  xml: {
+    book: OpacBook[]
+  }
+}
+
+/**
+ * Non-persistent storage for fetched books.
+ */
+let fetchedBooks: Book[] | null = null
+
+/**
+ * An object containing coverUrl and whether the cover is not generic (is generic = false).
+ */
+const bookCoverMap: Record<string, boolean> = {}
+
+/**
+ * Goes all over unchecked images a detects whether the book has a generic image cover.
+ */
+const checkCoverImages = async () => {
+  // eslint-disable-next-line no-restricted-syntax
+  for (const book of fetchedBooks ?? []) {
+    if (!book.coverUrl) {
+      // eslint-disable-next-line no-continue
+      continue
+    }
+    try {
+      // Image is already checked.
+      if (book.coverUrl in bookCoverMap) {
+        // eslint-disable-next-line no-continue
+        continue
+      }
+      // `url-file-size` gets only filesize information from the server, there's no need to download
+      // the whole image.
+      // eslint-disable-next-line no-await-in-loop
+      const coverFileSize = await ufs(book.coverUrl)
+
+      // Checking image size in bytes is a reliable way how to detect a generic cover.
+      bookCoverMap[book.coverUrl] = coverFileSize !== 9149
+
+      // eslint-disable-next-line no-empty
+    } catch (error) {}
+  }
+}
+
+const fetchBooks = async () => {
+  // In localhost environments mock books are returned not to make unnecessary request as the service
+  // is restarted with each compilation.
+  if (process.env.NEXT_PUBLIC_IS_LOCALHOST === 'true') {
+    fetchedBooks = mockNewBooks
+    return
+  }
+
+  try {
+    const response = await fetch(bookNewsUrl)
+    const text = await response.text()
+    const opac: OpacResponse = XMLConvertor.xml2js(text, { compact: true }) as OpacResponse
+
+    fetchedBooks = opac.xml.book.map((book) => {
+      /* eslint-disable no-underscore-dangle */
+      const coverUrl = book?.coverURL?._text
+      return {
+        title: book?.title?._text,
+        url: book?.recURL?._text,
+        ...(coverUrl === ' --- ' ? {} : { coverUrl }),
+      }
+      /* eslint-enable no-underscore-dangle */
+    })
+    // Check is called asynchronously without waiting for a result.
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    checkCoverImages()
+    // eslint-disable-next-line no-empty
+  } catch (error) {}
+}
+
+let refreshIntervalTimer: NodeJS.Timer | null = null
+
+export interface NewBooksResponse {
+  books: Book[]
+  total: number
+}
+
+export const getNewBooks = async ({
+  page,
+  pageSize,
+}: {
+  page?: number
+  pageSize?: number
+}): Promise<NewBooksResponse> => {
+  if (!refreshIntervalTimer) {
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    refreshIntervalTimer = setInterval(() => fetchBooks(), 3_600_000)
```

Comment:
Maybe add comment how much is it in more readable unit such as minutes/hours?

Reply:
Added comment.